File: LMIS/LMIS.Web/Modules/BookManage/Publishers/PublishersColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace LMIS.BookManage.Columns;

[ColumnsScript("BookManage.Publishers")]
[BasedOnRow(typeof(PublishersRow), CheckNames = true)]
public class PublishersColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public long PublisherId { get; set; }
    [EditLink]
    public string PublisherName { get; set; }
    public string Contact { get; set; }
    public string ContactPhone { get; set; }
    public string Email { get; set; }
    public DateTime CreateTime { get; set; }
    public DateTime UpdateTime { get; set; }
}

File: LMIS/LMIS.Web/Modules/ServerTypes/BookManage/PublishersColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { PublishersRow } from "./PublishersRow";

export interface PublishersColumns {
    PublisherId: Column<PublishersRow>;
    PublisherName: Column<PublishersRow>;
    Contact: Column<PublishersRow>;
    ContactPhone: Column<PublishersRow>;
    Email: Column<PublishersRow>;
    CreateTime: Column<PublishersRow>;
    UpdateTime: Column<PublishersRow>;
}

export class PublishersColumns extends ColumnsBase<PublishersRow> {
    static readonly columnsKey = 'BookManage.Publishers';
    static readonly Fields = fieldsProxy<PublishersColumns>();
}